examples/app/random.rs

```diff
@@ -0,0 +1,96 @@
+use bevy::prelude::*;
+use rand::{prelude::IteratorRandom, rngs::SmallRng, Rng, SeedableRng};
+
+/// This example illustrates how to use entropy to control randomness in bevy.
+/// We randomly choose a coin to toss and record the result.
+///
+/// Because all random number generators are seeded from the same "world seed",
+/// this program is deterministic.
+struct Coin;
+
+struct Toss(SmallRng);
+
+#[derive(Debug, PartialEq)]
+enum Face {
+    Heads,
+    Tails,
+}
+
+fn main() {
+    // This is a bad seed but is used for illustrative purposes.
+    //
+    // Normally you would do one of the following:
+    //   1. Get a good random seed out-of-band and hardcode it in the source.
+    //   2. Dynamically call to the OS and print the seed so the user can rerun
+    //      deterministically.
+    //   3. Dynamically call to the OS and share the seed with the server so the
+    //      client and server deterministically execute together.
+    //   4. Load the seed from a server so the client and server deterministically
+    //      execute together.
+    let world_seed: [u8; 32] = [1; 32];
```

Comment:
IMO we need a "what is a random seed and what does it do" explanation here for complete beginners. The array of 32 integers will be very baffling to new programmers.